### DISCUSSION TITLE
Back4App - Was Working - Now Stopped

### DISCUSSION THREAD

**Discussion by JAA17** [peers/peerjs#1222] [Q&A]
PeerJS Server was working on Back4App where you could get the message, but now even that has stopped working with the new DockerFile.

Back4App uses the Dockerfile to set up.

What I have done.

1. On the 'My Apps' page when signed into Back4App.com, click 'Build New App'.
2. Click the 'Containers as a Service'.
3. You will need to connect your Github account (must be the same email). Once you do, your Github projects will appear. Then click 'Select' for your copy of PeerJS Server in your Github account.
4. Fill out the form, eg App Name: mypeerjsserver, then the rest as defaults.
5. Click 'Create App'
6. It will create it using the Dockerfile in the PeerJS Server.

Once it becomes active, you get a url, eg 'mypeerjsserver6-xxxxxx.b4a.run'.

If you type this into your browser, you ***used*** to get your response message and everything was working fine.

Now we do not even get that.

That is the finish of the issue, however while it was working I could not work out the following:

---

I just cannot connect to my Back4App PeerJS Server in Client Code.

You can use a third-party code to test:

https://jsfiddle.net/NathanFriend/q6tch0az/39/

The line that connects is:

const peer = new Peer(id);
However, this only connects to the general server run by PeerJS. To connect it to my server, you have to edit this line to:

const peer = new Peer(id, {host: "mypeerjsserver6-xxxxxx.b4a.run", port: 443, secure: true});
I think the port has to be 443 because back4app is a secure host. I think I have the host field set correctly.

Do I need to do edits in the dockerfile to address the possible 443 port? How do I do that in Github?

I am so close, but so far away.

**Comment by jonasgloning** [comment]
I think they have some problems with multi-stage builds. But the following works:

Create an empty Github repository and put there a `Dockerfile` with the following content[^1]:
```Dockerfile
FROM peerjs/peerjs-server:1.0.0
```

Then connect with `host`, `secure`, and `port` set. 
```ts
const peer = new Peer(id, {host: "mypeerjsserver6-xxxxxx.b4a.run", port: 443, secure: true});
```


[^1]: You can of course change the `:tag` to something more broad, if you want.

**Comment by JAA17** [comment]
Hi jonasgloning. Your answers gets a long way, but still does not work.

I have done what you have said with the Dockerfile.

Back4App then gives a url, eg "https://mypeerjsserver10-XXXXX.b4a.run". The XXXXX are my user name.

This url gives the correct response:

{"name":"PeerJS Server","description":"A server side element to broker connections between PeerJS clients.","website":"https://peerjs.com/"}

How do I connect to this Back4App server from my javascript client code?

I thought it would be:

const peer = new Peer(id, {host: "mypeerjsserver10-XXXXX.b4a.run", port: 443, secure: true});

But this does not connect.

Is it the host, port or secure? Do I need a path? If so, what path?

If you want to try this, it is simple to set up.

What I have done:

**Setting Up Your Own Github PeerJS Server Repository**

So set up an empty Github project with just a Dockerfile with the single line contents:

FROM peerjs/peerjs-server:1.0.0

**Setting Up Your Back4App Container**

Back4App has made this very simple as they process this with the Dockerfile in your Github repository.

    1 On the 'My Apps' page when signed into Back4App.com, click 'Build New App'.
    2 Click the 'Containers as a Service'.
    3 You will need to connect your Github account (must be the same email). Once you do, your Github projects will appear. Then click 'Select' for your copy of PeerJS Server in your Github account.
    4 Fill out the form, eg App Name: mypeerjsserver, then the rest as defaults.
    5 Click 'Create App'

When it loads everything, in the top left you will get a green tick and the word 'Available'. Underneath will be your url for the app, eg mypeerjsserver10-XXXXX.b4a.run.

If you click on that url you will direct to a webpage with the following:

{"name":"PeerJS Server","description":"A server side element to broker connections between PeerJS clients.","website":"https://peerjs.com/"}

So it is all working.

Note the url is an https so it is secure.

**Testing the PeerJS Container**

There is a Jsfiddle by someone else.

https://jsfiddle.net/NathanFriend/q6tch0az/39/

He uses the general server run by the PeerJS creators.

He connects to that with the line:

const peer = new Peer(id);

To connect it to your server, you have to edit this line to something like:

const peer = new Peer(id, {host: "mypeerjsserver10-XXXXX.b4a.run", port: 443, secure: true});

But this obviously does not work.

I am close, but still a long way off.

**Reply by JAA17** [reply]
I still cannot get this to connect. Please help someone.

I have done (as suggested)

Create an empty Github repository and put there a Dockerfile with the following content[1](https://github.com/peers/peerjs-server/discussions/357#user-content-fn-1-10a1f368305da67620a7b998313e6937):

`FROM peerjs/peerjs-server:1.0.0`

But I cannot do the following:

Then connect with host, secure, and port set.

`const peer = new Peer(id, {host: "mypeerjsserver6-xxxxxx.b4a.run", port: 443, secure: true});`

Has anyone tried running PeerJS Server on Back4App?

They have free accounts for those who want to try.

**Reply by JAA17** [reply]
Also, what does this mean? Who is the 'they' in this sentence?

'I think they have some problems with multi-stage builds.'
